sklearn/linear_model/_glm/_newton_solver.py

```diff
@@ -516,3 +532,460 @@ def inner_solve(self, X, y, sample_weight):
                 )
             self.use_fallback_lbfgs_solve = True
             return
+
+
+class NewtonLSMRSolver(NewtonSolver):
+    """LSMR based Newton solver.
+
+    The inner solver uses LSMR [1] after the Newton update is cast into the iteratively
+    reweighted least squares (IRLS) formulation. This means
+
+        H @ coef_newton = -G
+
+    with
+
+        G = X.T @ g + l2_reg_strength * coef
+        H = X.T @ diag(h) @ X + l2_reg_strength * P
+        g = loss.gradient = pointwise gradient
+        h = loss.hessian = pointwise hessian
```

Comment:
I think "Hessian" should always be capitalized in comments and docstrings.

```suggestion
        h = loss.hessian = pointwise Hessian
```

Reply:
In a loooot of other places, we simply use "hessian" not "Hessian".